src/components/ConnectCeramicButton.tsx

```diff
@@ -0,0 +1,236 @@
+import { FC, useCallback, useEffect, useState } from "react";
+import {
+  Alert,
+  Box,
+  Button,
+  CircularProgress,
+  Typography,
+} from "@mui/material";
+import {
+  EthereumAuthProvider,
+  useViewerConnection,
+  useViewerRecord,
+} from "@self.id/framework";
+import {
+  AddressBookEntry,
+  addressBookSelectors,
+  addressBookSlice,
+} from "../redux/slices/addressBook.slice";
+import { useAppDispatch, useAppSelector } from "../redux/hooks";
+
+type Contact = {
+  name: string;
+  wallets: Wallet[];
+};
+type Wallet = {
+  network: string;
+  walletAddress: string;
+};
+
+// Connect to Ceramic with injected wallet
+async function connectWallet(connect: Function) {
+  try {
+    const accounts = await (global as any).ethereum.request({
+      method: "eth_requestAccounts",
+    });
+    // @ts-ignore
+    return connect(new EthereumAuthProvider(window.ethereum, accounts[0]));
+  } catch (error) {
+    console.log("Error connecting wallet", error);
+  }
+}
+
+// Sync state with Ceramic.
+// Another approach would be to listen to changes in addressBookEntries but
+// wasn't able to make work in a good way.
+export function useAdressBookSync() {
+  // @ts-ignore
+  const record = useViewerRecord("myAddressBook");
```

Comment:
The data is encrypted and secured on Ceramic's end so the "viewer record" is only visible to the authenticated identity, right?

I would appreciate a link where this is stated explicitly as well :pray: [carlbarrdahl](https://github.com/carlbarrdahl)